src/System.Private.ServiceModel/tools/IISHostedWcfService/App_code/testhosts/TestServiceHostBase.cs

```diff
@@ -0,0 +1,92 @@
+﻿//  Copyright (c) Microsoft Corporation.  All Rights Reserved.
+using System;
+using System.ServiceModel;
+using System.ServiceModel.Channels;
+using System.ServiceModel.Description;
+
+namespace WcfService
+{
+    public abstract class TestServiceHostBase<ContractType>: ServiceHost
+    {
+        protected static int SixtyFourMB = 64 * 1024 * 1024;
+        protected abstract string Address { get; }
+
+        protected abstract Binding GetBinding();
+       
+
+        public TestServiceHostBase(Type serviceType, params Uri[] baseAddresses)
+            : base(serviceType, baseAddresses)
+        {
+            ServiceEndpoint endpoint = this.AddServiceEndpoint(typeof(ContractType), GetBinding(), Address);
+        }
+
+        //Overriding ApplyConfiguration() allows us to 
+        //alter the ServiceDescription prior to opening
+        //the service host. 
+        protected override void ApplyConfiguration()
+        {
+            //First, we call base.ApplyConfiguration()
+            //to read any configuration that was provided for
+            //the service we're hosting. After this call,
+            //this.ServiceDescription describes the service
+            //as it was configured.
+            base.ApplyConfiguration();
+
+            ServiceDebugBehavior debugBehavior = this.Description.Behaviors.Find<ServiceDebugBehavior>();
+            if (debugBehavior == null)
+            {
+                debugBehavior = new ServiceDebugBehavior();
+                this.Description.Behaviors.Add(debugBehavior);
+            }
+            debugBehavior.IncludeExceptionDetailInFaults = true;
+            //Now that we've populated the ServiceDescription, we can reach into it
+            //and do interesting things (in this case, we'll add an instance of
+            //ServiceMetadataBehavior if it's not already there.
+            ServiceMetadataBehavior mexBehavior = this.Description.Behaviors.Find<ServiceMetadataBehavior>();
```

Comment:
What is the purpose of adding this mexBehavior to all our service hosts?
I didn't think we had any way to access mex info from the client. 
Also, we don't support named pipes